lib/util/utimes.js

```diff
@@ -1,17 +1,33 @@
 'use strict'
 
-const fs = require('graceful-fs')
+const fs = require('../fs')
+const u = require('universalify').fromPromise
 
-function utimesMillis (path, atime, mtime, callback) {
+async function utimesMillis (path, atime, mtime, cb) {
   // if (!HAS_MILLIS_RES) return fs.utimes(path, atime, mtime, callback)
-  fs.open(path, 'r+', (err, fd) => {
-    if (err) return callback(err)
-    fs.futimes(fd, atime, mtime, futimesErr => {
-      fs.close(fd, closeErr => {
-        if (callback) callback(futimesErr || closeErr)
-      })
-    })
-  })
+  const fd = await fs.open(path, 'r+')
+
+  let futimesErr = null
+  try {
+    await fs.futimes(fd, atime, mtime)
+  } catch (e) {
+    futimesErr = e
+  }
+
+  let closeErr = null
+
+  try {
+    await fs.close(fd)
+  } catch (e) {
+    closeErr = e
+  }
+
+  if (futimesErr) {
+    throw futimesErr
+  }
```

Comment:
We can achieve the same behavior a little more simply with `try..finally`.
```suggestion
  let closeErr
  try {
    await fs.futimes(fd, atime, mtime)
  } finally {
    // put this in a try..catch so the error from futimes is thrown preferentially to any error here
    try {
      await fs.close(fd)
    } catch (e) {
      closeErr = e
    }
  }
```

Reply:
Changed in [`1ce5d34` (#1021)](https://github.com/jprichardson/node-fs-extra/pull/1021/commits/1ce5d348c426ea20c5006389477e163211898b22).